compaction/compaction_manager.cc

```diff
@@ -384,6 +395,15 @@ compaction_manager::run_with_compaction_disabled(table* t, std::function<future<
     co_return;
 }
 
+compaction_manager::task::task(compaction_manager::task&& o) noexcept
+        : compacting_cf(std::exchange(o.compacting_cf, nullptr))
+        , type(o.type)
+        , compaction_state(o.compaction_state)
+        , gate_holder(std::move(o.gate_holder))
```

Comment:
do we really need this move ctor? also it's missing a couple of fields like task::compaction_done

Reply:
It's just for debuggibg.
Actually since we're keeping a shared pointer of the task I think we can just delete it

Reply:
Done in v5